pkg/kn/commands/revision/revision_list.go

```diff
@@ -39,7 +48,18 @@ func NewRevisionListCommand(p *commands.KnParams) *cobra.Command {
 			if err != nil {
 				return err
 			}
-			revision, err := client.Revisions(namespace).List(v1.ListOptions{})
+			listOptions := v1.ListOptions{}
+			if cmd.Flags().Changed("service") {
+				service := cmd.Flag("service").Value.String()
+				// Ensure requested service exist
+				_, err := client.Services(namespace).Get(service, v1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				listOptions.LabelSelector = labels.Set(
+					map[string]string{serving.ConfigurationLabelKey: service}).String()
```

Comment:
Shouldn't this be `serving.ServiceLabelKey` ? Don't really see how this is tested in the tests, too ...

Reply:
Hmm, I checked an actual revision and both the keys in labels are holding service name
```
                "labels": {
                    "serving.knative.dev/configuration": "svc1",
                    "serving.knative.dev/configurationGeneration": "1",
                    "serving.knative.dev/service": "svc1"
                },
```

@rhuss : I will need to check among 2 labels, which one we should filter against.

in tests [here](https://github.com/knative/client/pull/194/files#diff-482bb3cf4550490e7cb84ffc5ddc760fR140).

Reply:
But isn't the service label the authorative answer for which revision belongs to which service and the configuration is just (by accident) named like the service because it has been created implicitely ? Or does the configuration alway be named like the service ? (I don't think so)